src/python/pants/pantsd/pants_daemon.py

```diff
@@ -65,6 +72,10 @@ def isatty(self):
   def fileno(self):
     return self._handler.stream.fileno()
 
+  @property
+  def buffer(self):
+    return self
```

Comment:
I would almost definitely prefer to subclass `_LoggerStream` e.g. `_LoggerByteStream` (probably the other way around, looking at the `ensure_text()` call) and return an instance of that for `_LoggerStream#buffer`.